packages/cli-repl/src/completer.ts

```diff
@@ -34,49 +67,89 @@ function completer(mdbVersion: string, line: string): [string[], string] {
+      // aggregation cursor completions
       if (splitLine[2].includes('aggregate')) {
-        const hits = filterComplete(mdbVersion, aggCursorComplete, elToComplete, splitLine);
+        const hits = filterShellAPI(
+          mdbVersion, AGG_CURSOR_COMPLETIONS, elToComplete, splitLine);
         return [hits.length ? hits: [], line];
       }
-      const hits = filterComplete(mdbVersion, collCursorComplete, elToComplete, splitLine);
+      // collection cursor completions
+      const hits = filterShellAPI(
+        mdbVersion, COLL_CURSOR_COMPLETIONS, elToComplete, splitLine);
       return [hits.length ? hits : [], line];
     }
-    // if splitLine[2] === 'aggregate' && splitLine[2].contains('({'), suggest
-    // aggregation operators from 'ace-autocompleter'
 
-    // if splitLine[2].contains(any of the collComplete), and
-    // .contains('({'), suggest query operators from 'ace-autocompleter'
+    // complete aggregation and collection  queries/stages
+    if (splitLine[2].includes('([') || splitLine[2].includes('({')) {
+      let expressions;
+      if (splitLine[2].includes('aggregate')) {
+        // aggregation needs extra accumulators to autocomplete properly
+        expressions = BASE_COMPLETIONS.concat(accumulators(
+          elToComplete, mdbVersion));
+      } else {
+        // collection quering just needs MATCH COMPLETIONS
+        expressions = MATCH_COMPLETIONS;
+      }
+      // split on {, as a stage/query will always follow an open curly brace
+      const splitQuery = line.split("{");
+      const prefix = splitQuery.pop().trim();
+      const command = line.split(prefix).shift();
+      const hits = filterQueries(mdbVersion, expressions, prefix, command);
+      return [hits.length ? hits: [], line]
+    }
 
-    // else:
-    const hits = filterComplete(mdbVersion, collComplete, elToComplete, splitLine);
+    const hits = filterShellAPI(
+      mdbVersion, COLL_COMPLETIONS, elToComplete, splitLine);
     return [hits.length ? hits: [], line];
   } else if (firstLineEl.includes('sh')) {
-    const hits = filterComplete(mdbVersion, shardComplete, elToComplete, splitLine);
+    const hits = filterShellAPI(
+      mdbVersion, SHARD_COMPLETE, elToComplete, splitLine);
     return [hits.length ? hits : [], line];
   } else if (firstLineEl.includes('rs')) {
-    const hits = filterComplete(mdbVersion, rsComplete, elToComplete, splitLine);
+    const hits = filterShellAPI(
+      mdbVersion, RS_COMPLETIONS, elToComplete, splitLine);
     return [hits.length ? hits : [], line];
   }
 
   return [[line], line];
 }
 
-// TODO: @lrlna this should also take this.shellApi.version of sorts and also
-// match on version:
-// semver.gte(version, .version)
-function filterComplete(mdbVersion: string, completions: object, toMatchTo: string, split?:
-                        string[]) {
+// stage completions based on current stage string.
+function accumulators(stage: string, mdbVersion: string) {
```

Comment:
what about a name that is more inline with the comment? like `getStageCompletions` or `stageCompletionsAccumulators`.